content/blog/accessing-dtap-in-pods.md

```diff
@@ -0,0 +1,264 @@
+cd $HOME/hadoop                                                                           # Move directory to hadoop
+```
+
+
+
+### Configure the required environment
+
+In ```$HADOOP_HOME/etc/hadoop/hadoop-env.sh``` file, assign the following environment variables (```$JAVA_HOME```, ```$HADOOP_HOME```, ```$HADOOP_CLASSPATH```):
+
+```bash
+# These two variables is needed for HDFS command. Located at line 54, 58.
+export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64/
+export HADOOP_HOME=$HOME/hadoop
+
+# This variable is DataTap specific. Located at line 126.
+export HADOOP_CLASSPATH=$HADOOP_CLASSPATH:$HADOOP_HOME/lib/:/opt/bdfs/bluedata-dtap.jar
+```
+
+In ```$HADOOP_HOME/etc/hadoop/core-site.xml``` file, configure Hadoop with the following values:
+
+```xml
+<configuration>
+  <property>
+    <name>fs.dtap.impl</name>
+    <value>com.bluedata.hadoop.bdfs.Bdfs</value>
+  </property>
+
+  <property>
+    <name>fs.AbstractFileSystem.dtap.impl</name>
+    <value>com.bluedata.hadoop.bdfs.BdAbstractFS</value>
+  </property>
+
+  <property>
+    <name>fs.dtap.impl.disable.cache</name>
+    <value>false</value>
+  </property>
+</configuration>
+```
+
+### Alternative
+
+I have prepared an example configuration file on [Github](https://github.com/helloezmeral/hpe-binary/tree/main/hadoop-dtap-config). If your Hadoop does not have a special configuration, you can simply download and replace your existing configuration file.
+
+## Test your HDFS command
+
+Here are some common commands used to interact with DataTap.
+
+```bash
+# bash: Current working directory -> $HADOOP_HOME
+
+# Check the version of the Hadoop.
+bin/hadoop version
+
+# List the files from the default TenantStorage Data Source.
+bin/hdfs dfs -ls dtap://TenantStorage/
+
+# Make new directory user in dtap://TenantStorage/.
+bin/hdfs dfs -mkdir dtap://TenantStorage/user
+
+# Move the text files helloworld.txt to "cenz" folder.
+bin/hdfs dfs -put helloworld.txt dtap://TenantStorage/cenz
+bin/hdfs dfs -put -f helloworld.txt dtap://TenantStorage/cenz # force replacement
+
+# Concatenate a file in dtap.
+bin/hdfs dfs -cat dtap://TenantStorage/cenz/helloworld.txt
+
+# Remove a file in dtap.
+bin/hdfs dfs -rm dtap://TenantStorage/cenz/helloworld.txt
+```
+
+> Tips:
+> 
+> - to get rid of the file path ```bin/```, we can add the Hadoop ```bin``` and ```sbin``` file to ```$PATH```
+> 
+> ```
+> export HADOOP_HOME=$HOME/hadoop
+> export PATH=$PATH:$HADOOP_HOME:$HADOOP_HOME/sbin:$HADOOP_HOME/bin
+> ```
+
+
+> Reference: 
+> [Hadoop File System Shell Document](https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-common/FileSystemShell.html)
+
+# Access DataTaps using pyspark
+
+## Introduction
+
+PySpark is an interface for Apache Spark in Python. Apache Spark is a unified analytics engine for big data processing, with built-in modules for streaming, SQL, machine learning and graph processing. Apache Spark can access data from HDFS and, with extension, file systems managed by DataTap. 
```

Comment:
"...and, with extension, file systems..." should be either "...and, with extension**s,** file systems..." OR "and, with **the** extension, file systems..."